DPPP/BDABuffer.cc

```diff
@@ -0,0 +1,216 @@
+// Copyright (C) 2020
+// ASTRON (Netherlands Institute for Radio Astronomy)
+// P.O.Box 2, 7990 AA Dwingeloo, The Netherlands
+//
+// This file is part of the LOFAR software suite.
+// The LOFAR software suite is free software: you can redistribute it and/or
+// modify it under the terms of the GNU General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The LOFAR software suite is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with the LOFAR software suite. If not, see <http://www.gnu.org/licenses/>.
+
+#include "BDABuffer.h"
+
+#include <algorithm>
+#include <limits>
+
+namespace DP3 {
+  namespace DPPP {
+
+    BDABuffer::Row::Row(double time,
+                        double interval,
+                        rownr_t row_nr,
+                        std::size_t baseline_nr,
+                        std::size_t n_channels,
+                        std::size_t n_correlations,
+                        std::complex<float>* data,
+                        bool* flags,
+                        float* weights,
+                        bool* full_res_flags,
+                        const double *const uvw)
+    : time_(time)
+    , interval_(interval)
+    , row_nr_(row_nr)
+    , baseline_nr_(baseline_nr)
+    , n_channels_(n_channels)
+    , n_correlations_(n_correlations)
+    , data_(data)
+    , flags_(flags)
+    , weights_(weights)
+    , full_res_flags_(full_res_flags)
+    , uvw_{ uvw ? uvw[0] : std::numeric_limits<double>::quiet_NaN(),
+            uvw ? uvw[1] : std::numeric_limits<double>::quiet_NaN(),
+            uvw ? uvw[2] : std::numeric_limits<double>::quiet_NaN() }
+    {}
+
+    BDABuffer::BDABuffer(const std::size_t pool_size,
+                         const Fields& fields)
+    : data_()
```

Comment:
This is not the google syntax for class member initialization (comma should be after variable), see https://google.github.io/styleguide/cppguide.html#Constructor_Initializer_Lists
but I hope the linting tool will correct this once we turn that on